File: src/CSP-AF/CSPCore/CSPGetObjectId.cs
using System;
using System.Linq;
using System.Collections.Generic;

using CloudSuitePortal;
using CloudSuitePortal.Services;

namespace actions.CSP
{
    class CSPGetObjectId
    {
        public string run(Dictionary<string, object> Params)
        {
            string prefix = (string)Params["Prefix"];
            string function = (string)Params["Function"];
            string fileName = (string)Params["File Name"];
            string filePath = CSPLogger.ProjectDirectoryPath() + "\\" + fileName;
            
            var validFunctions = new string[] {"get", "increment"};
            if (!validFunctions.Contains(function.ToLower())) { return null; }

            var objectId = prefix;
            var logger = new CSPLogger(filePath);
            var id = logger.NumberOfLines();

            if (function.ToLower() == "get") { id -= 1; }

            return objectId + id;
        }
    }
}

File: src/CSPAF/CSPCore/CSPGetObjectId.java
package CSPAF.CSPCore;

import java.util.*;

import com.infor.cloudsuite.portal.core.helpers.*;
import com.infor.cloudsuite.portal.core.services.*;

class CSPGetObjectId
{
    public String run(HashMap<String, Object> params) throws Exception
    {
       	String prefix = params.get("Prefix").toString();
        String function = params.get("Function").toString();
        String fileName = params.get("File Name").toString();
        String filePath = CSPLogger.projectDirectoryPath() + "\\" + fileName;
            
        List<String> validFunctions = Arrays.asList("get", "increment");
        if (!validFunctions.contains(function.toLowerCase())) { return null; }
        
        CSPLogger<String> logger = new CSPLogger<String>(filePath, String.class);
        logger.create();
        
        CSPDebugHelper.debugPrint("Fetching Ids...");
        
        List<String> ids = logger.fetchObjects();
        
        int numberOfIds = ids.size();
        
        String id = null;
		
      	if (function.toLowerCase().equals("increment"))
        { 
            String suffix = numberOfIds > 0 ? "" + numberOfIds : "0";
           	id = prefix + suffix;
           	logger.log(id);
           	CSPDebugHelper.debugPrint("Logging id: " + id);
       	}
        else if (function.toLowerCase().equals("get") && numberOfIds > 0) {
            id = ids.get(numberOfIds - 1);
        }
        
        return id;
    }
}